src/diffusers/models/attention.py

```diff
@@ -12,12 +12,36 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import math
-from typing import Optional
+import os
+from inspect import isfunction
+from typing import Any, Optional
 
 import torch
 import torch.nn.functional as F
 from torch import nn
 
+from diffusers.utils.import_utils import is_xformers_available
+
+
+if is_xformers_available():
+    import xformers
+    import xformers.ops
+
+    _USE_MEMORY_EFFICIENT_ATTENTION = int(os.environ.get("USE_MEMORY_EFFICIENT_ATTENTION", 0)) == 1
+else:
+    xformers = None
+    _USE_MEMORY_EFFICIENT_ATTENTION = False
```

Comment:
Thank you for fixing this @MatthieuTPHR :)

@patrickvonplaten users now have to set `USE_MEMORY_EFFICIENT_ATTENTION=1` if they want to use xformers' attention. Is it fine like this? or would it better to have something like `enable_mem_efficient_attention`, like it's the case for [attention slicing](https://github.com/matthieutphr/diffusers/blob/e82a5203f5e4f6cb04ed6b127da88abe9fdbf8f2/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L94-L111)? (which would allow for nice error messages like in [here](https://github.com/matthieutphr/diffusers/blob/e82a5203f5e4f6cb04ed6b127da88abe9fdbf8f2/src/diffusers/models/unet_blocks.py#L355-L368), in case of absence of xformers library or bad kernels compilation)

Reply:
I think a `enable_attention_xformers` would make a lot of sense :-)

Reply:
Agree with this, let's add a `enable_attention_xformers` method.

Reply:
Working on adding this